pkg/utils/k8sutil.go

```diff
@@ -784,10 +784,12 @@ func GetLocalHostname(config *rest.Config, funcName string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-
-	host, _, err := net.SplitHostPort(url.Host)
-	if err != nil {
-		return "", err
+	host := url.Host
+	if strings.Contains(url.Host, ":") {
+		host, _, err = net.SplitHostPort(url.Host)
```

Comment:
This is still not correct on ipv6-literal URLs fwiw, which may include embedded colons in the "host" portion of the URL.  Even better fix following :P